hera_cal/lst_stack/metrics.py

```diff
@@ -64,11 +66,48 @@ def get_nightly_predicted_variance(
     dtdf = (stack.dt * stack.df).to_value("")
 
     gf = stack.get_flags(bl)
-    per_day_expected_var = np.abs(auto / dtdf / stack.get_nsamples(bl))
+    with np.errstate(divide='ignore'):
+        # Some nsamples can be zero, so we ignore warnings from that.
+        # Also note that in the stack, inpainted samples have -nsamples, rather than
+        # zero. These will thus return a finite expected variance, even though they
+        # technically have no unflagged samples.
+        per_day_expected_var = np.abs(auto / dtdf / stack.get_nsamples(bl))
     per_day_expected_var[gf] = np.inf
     return per_day_expected_var
 
 
+def get_nightly_predicted_variance_stack(
```

Comment:
this duplicates some of the functionality in https://github.com/HERA-Team/hera_cal/blob/e56a6868427fa956ac6035f339d9ec8f2e107b30/hera_cal/noise.py#L71 ... is it possible to reuse that? or not worth it?

Reply:
I think it's fine for now.